draft-ietf-rtcweb-mdns-ice-candidates.md

```diff
@@ -296,8 +346,36 @@ IP addresses that this specification is designed to hide. The use of
 registered mDNS hostnames SHOULD be scoped by origin, and SHOULD have the
 lifetime of the page.
 
+Determination of Address Privacy
+--------------------------------
+
+Naturally, an address that is already exposed to the Internet does not need to
+be protected by mDNS, as it can be trivially observed by the web server or
+remote endpoint. However, determining this ahead of time is not straightforward;
+while the fact that an IPv4 address is private can sometimes be inferred by its
+value, e.g., whether it is an {{RFC1918}} address, the reverse is not
+necessarily true. IPv6 addresses present their own complications, e.g.,
+private IPv6 addresses as a result of NAT64 {{RFC6146}}.
+
+Instead, the determination of whether an address is public can be reliably made
+as part of the ICE gathering process, namely, if the query to the
+STUN {{RFC5389}} server returns the same value as the local address. This can
+be done for both IPv4 and IPv6 local addresses, provided that the application
+has configured both IPv4 and IPv6 STUN servers. Implementations MAY cache this
```

Comment:
IIUC, this means the first gathering would add some latency for host candidates because we need to wait for STUN to determine their sensitivity. Should we define/recommend default options for concealment if this is not done by the implementation or techniques like STUN fail/delay too much?